react/week3/react1-fetchapi-week3/src/components/TodoList.jsx

```diff
@@ -0,0 +1,183 @@
+import Timer from "./Timer.jsx";
+
+const TODO_URL =
+  "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw";
+
+const InputData = ({ addTodo }) => {
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    addTodo(description, deadline);
+    setDescription("");
+    setDeadline("");
+  };
+
+  const minDate = () => {
+    let newDate = new Date();
+    let date = newDate.getDate();
+    //let date=("0" + newDate.getDate()).slice(-2);
+    let month = newDate.getMonth() + 1;
+    //let month = ("0" + (newDate.getMonth() + 1)).slice(-2);
+    //let year = newDate.getFullYear();
+     let currentDate = newDate.getFullYear() + "-" + month + "-" + date;
+  //let currentDate= date+ "-" + month + "-" + newDate.getFullYear();
+    return currentDate;
+  };
+
+
+
+  return (
+    <div>
+      <form onSubmit={onSubmit}>
+        <div>
+          <label htmlFor="description">Todo Description:</label>
+          <input
+            type="text"
+            required
+            id="description"
+            value={description}
+            placeholder="Description"
+            onChange={(e) => {
+              const value = e.target.value;
+              console.log(value);
+              setDescription(value);
+            }}
+          />
+        </div>
+        <div>
+          <label htmlFor="deadline">Deadline:</label>
+          <input
+            type="date"
+            required 
+            id="deadline"
+            min={minDate()}
+            value={deadline}
+            placeholder="Deadline"
+            onChange={(e) => {
+              const value = e.target.value;
+              setDeadline(value);
+            }}
+          />
+        </div>
+        <button type="submit">Add Todo</button>
+      </form>
+    </div>
+  );
+};
+
+function Todo({ description, deadline, onDelete, onUpdate, id }) {
+  const [check, setCheck] = useState(false);
+  const [edit, setEdit] = useState(false);
+  const [newDescription, setNewDescription] = useState(description);
+  const onCheck = () => {
+    return setCheck(!check);
+  };
+
+  const onEdit = () => {
+    setEdit(!edit);
+    onUpdate(id, newDescription);
+  };
+  return (
+    <div className="todo">
+      <div className="items">
+        {edit ? (
+          <input
+            type="text"
+            required
+            value={newDescription}
+            onChange={(e) => setNewDescription(e.target.value)}
+          />
+        ) : (
+          <h3 style={{ textDecoration: check ? "line-through" : "none" }}>
+            {id} {description} {deadline}:
+          </h3>
+        )}
+        <input type="checkbox" checked={check} onChange={onCheck} />
+      </div>
+      <button onClick={onDelete}>Delete</button>
+      <button onClick={onEdit}>{edit ? "Update" : "Edit"}</button>
+    </div>
+  );
+}
+const Border = ({ children }) => {
```

Comment:
There is not really a reason to do thus border component as the only thing it is is a div with some styling, you might aswell put that in the component it is needed in.

Reply:
Hai @rallelind thanks for the time to review my code it is really helpful to learn more.
sorry to say.  I really did not understand the question can I delete this border component or replace it in some other place? thank you.

Reply:
Sorry i first see your comment now. But yes you can. The border component you have done is just a div with a className so instead just add the div with that className in the component where it is needed. I will add a comment in the comment i am talking about.